docs/user-guide/private-repositories.md

```diff
@@ -41,24 +41,32 @@ The Argo CD UI don't support configuring SSH credentials. The SSH credentials ca
 argocd repo add git@github.com:argoproj/argocd-example-apps.git --ssh-private-key-path ~/.ssh/id_rsa
 ```
 
-## Self-Signed Certificates
+## HTTPS Self-signed Certificate
 
-If you are using self-hosted Git hosting service with the self-signed certificate then you need to disable certificate validation for that Git host.
-Following options are available:
+We do not currently have first-class support for this. See [#1513](https://github.com/argoproj/argo-cd/issues/1513).
+
+As a work-around, you can customize your Argo CD image. See [#1344](https://github.com/argoproj/argo-cd/issues/1344#issuecomment-479811810)
 
-Add repository using Argo CD CLI and `--insecure-ignore-host-key` flag:
+## Unknown SSH Hosts
 
+If you are using a self-hosted Git service over SSH then you need to disable host verification for that Git host.
```

Comment:
> "then you need to disable host verification for that Git host."

This makes it seem they **_must_** go the route of disabling host verification, which means we are recommending they run in an insecure way, which we should never do. Lets rephrase this:

If you are using a privately hosted git service over SSH, then the following options are available: